homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,614 @@
+    ADD_OBSERVATION = str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+    SELECT_EDIT_OBSERVATION = "select_edit_observation"
+    EDIT_OBSERVATION = "edit_observation"
+    REMOVE_OBSERVATION = "remove_observation"
+
+    @staticmethod
+    def list_primary_steps() -> list[str]:
+        """Return a list of the values."""
+        li = [c.value for c in OptionsFlowSteps]
+        li.remove("init")
+        li.remove("edit_observation")
+        return li
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_PROBABILITY_THRESHOLD, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(
+            CONF_PRIOR, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Optional(CONF_DEVICE_CLASS): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=[cls.value for cls in BinarySensorDeviceClass],
+                mode=selector.SelectSelectorMode.DROPDOWN,
+                translation_key="binary_sensor_device_class",
+                sort=True,
+            ),
+        ),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): selector.TextSelector(),
+    }
+).extend(OPTIONS_SCHEMA.schema)
+
+SUBSCHEMA_BOILERPLATE = vol.Schema(
+    {
+        vol.Required(CONF_P_GIVEN_T): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(CONF_P_GIVEN_F): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(CONF_NAME): selector.TextSelector(),
+    }
+)
+
+ADD_ANOTHER_BOX_SCHEMA = vol.Schema({vol.Optional("add_another"): cv.boolean})
+
+STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, BINARY_SENSOR_DOMAIN, INPUT_BOLEAN_DOMAIN]
+            )
+        ),
+        vol.Required(CONF_TO_STATE): selector.TextSelector(
+            selector.TextSelectorConfig(
```

Comment:
Is there anyway to limit the states that can but inputted here to valid states for the entity selected above. This happens in the State Trigger ui in automations.

Reply:
In config flow it needs to be in separate forms so the second form is dynamic construction with the output from the first.

Reply:
OK, I think doing this would hurt more than it helped.